### PR TITLE
Fix dapr-tests on Windows  &&  scheduler statefulset yaml

### PR DESCRIPTION
**Pull request by cicoyle**
Trying to fix this CI failure: `unknown flag: --output` from https://github.com/dapr/dapr/actions/runs/9541403416/job/26294562718

### CODE REVIEW

**Comment by cicoyle** [comment]
I'm trying to reset some of the changes from [this PR](https://github.com/dapr/dapr/pull/7780). Will know once the test finishes

File: docker/docker.mk [hunk 1]
```diff
@@ -120,26 +120,26 @@ docker-build: check-docker-env check-arch
 	$(info Building $(DOCKER_IMAGE):$(DAPR_TAG) docker images ...)
 ifeq ($(TARGET_ARCH),$(TARGET_ARCH_LOCAL))
 ifeq ($(ONLY_DAPR_IMAGE),true)
-	$(DOCKER) build --output type=docker --build-arg PKG_FILES=* $(BUILD_ARGS) -f $(DOCKERFILE_DIR)/$(DOCKERFILE) $(BIN_PATH) -t $(DOCKER_IMAGE):$(BUILD_TAG)
```
Comment (artursouza) [inline docker/docker.mk:123]: This is needed for MacOS. It can be a variable. I am making these changes now.

**Comment by cicoyle** [comment]
That one e2e failure on kind looks to be transient: `dial tcp 172.18.0.2:31231: connect: connection refused`

**Comment by cicoyle** [comment]
Its running the e2e in azure, so looks to be thru the helm issue now

**Review by artursouza — APPROVED** [review]

**Pull request merged**